src/ape_ethereum/ecosystem.py

```diff
@@ -338,14 +338,49 @@ def decode_block(self, data: Dict) -> BlockAPI:
 
         return Block.parse_obj(data)
 
+    def _python_type_for_abi_type(self, abi_type: ABIType) -> Any[Type[Any]]:
+        # NOTE: An array can be an array of tuples, so we start with an array check
+        if str(abi_type.type).endswith("]"):
+            # remove one layer of the potential onion of array
+            new_type = "[".join(str(abi_type.type).split("[")[:-1])
+            # create a new type with the inner type of array
+            new_abi_type = ABIType(type=new_type, **abi_type.dict(exclude={"type"}))
+            # NOTE: type for static and dynamic array is a single item list
+            # containing the type of the array
+            return [self._python_type_for_abi_type(new_abi_type)]
+
+        if abi_type.components is not None:
+            return tuple(self._python_type_for_abi_type(c) for c in abi_type.components)
+
+        if abi_type.type == "address":
+            return AddressType
+
+        elif abi_type.type == "bool":
+            return bool
+
+        elif abi_type.type == "string":
+            return str
+
+        elif "bytes" in abi_type.type:
+            return bytes
+
+        elif "int" in abi_type.type:
+            return int
+
+        raise
```

Comment:
You can't use an empty raise like this!

```
>>> raise
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: No active exception to reraise
>>>
```

Please use an exception class

Reply:
(if there was an exception on the stack, it would work, but in this case, that isn't it)